tests/HubRestAPIClient.ts

```diff
@@ -0,0 +1,806 @@
+      });
+
+      it("can fetch a link by target ID", async function () {
+        const pageSize = 1;
+        const links = client.listLinksByTargetFid(6833, {
+          pageSize,
+        });
+        const link = await links.next();
+        expectDefinedNonNull(link.value);
+      });
+    });
+  });
+
+  describe("UserData API", function () {
+    describe("#getSpecificUserDataByFid", function () {
+      it("validates against OpenAPI spec", async function () {
+        const response = await client.apis.userData.getUserDataByFid({
+          fid: 6833,
+          userDataType: UserDataType.Username,
+        });
+        const validator = new OpenAPIResponseValidator({
+          components: apiSpec.components,
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          responses: apiSpec.paths["/v1/userDataByFid"].get.responses as any,
+        });
+        const errors = validator.validateResponse(200, response.data);
+        expect(errors, JSON.stringify(errors)).is.undefined;
+      });
+
+      it("filters on user data type", async function () {
+        const userDataUsername = await client.getSpecificUserDataByFid(
+          6833,
+          UserDataType.Username
+        );
+        expectDefinedNonNull(userDataUsername);
+        expect(userDataUsername.data.userDataBody.type).eq(
+          UserDataType.Username
+        );
+
+        const userDataPfp = await client.getSpecificUserDataByFid(
+          6833,
+          UserDataType.Pfp
+        );
+        expectDefinedNonNull(userDataPfp);
+        expect(userDataPfp.data.userDataBody.type).eq(UserDataType.Pfp);
+      });
+
+      it("returns null for nonexistent user data", async function () {
+        const userData = await client.getSpecificUserDataByFid(
+          userGaviFid,
+          UserDataType.Url
+        );
+        expect(userData).to.be.null;
+      });
+    });
+
+    describe("#listAllUserDataByFid", function () {
+      it("validates against OpenAPI spec", async function () {
+        const response = await client.apis.userData.getUserDataByFid({
+          fid: 6833,
+        });
+        expect((response.data as GetUserDataByFid200ResponseOneOf).messages).to
+          .not.be.empty;
+        const validator = new OpenAPIResponseValidator({
+          components: apiSpec.components,
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          responses: apiSpec.paths["/v1/userDataByFid"].get.responses as any,
+        });
+        const errors = validator.validateResponse(200, response.data);
+        expect(errors, JSON.stringify(errors)).is.undefined;
+      });
+
+      it("can fetch existing user data", async function () {
+        const pageSize = 1;
+        const userData = await client.listAllUserDataByFid(6833, { pageSize });
+        const singleData = await userData.next();
+        expectDefinedNonNull(singleData);
+      });
+    });
+  });
+
+  describe.skip("FIDs API", function () {
```

Comment:
add a comment to explain why this is skipped

Reply:
I think these hadn't been deployed yet until the latest hubble version went live. Now they pass 🎉